File: app/src/main/java/com/androidskeleton/module/ui/home/HomeContract.java
package com.androidskeleton.module.ui.home;


import com.androidskeleton.module.base.BasePresenter;
import com.androidskeleton.module.base.BaseView;

public interface HomeContract {
    interface View extends BaseView<Presenter> {

        void greetings(String message);
    }

    interface Presenter extends BasePresenter<View> {

        void sayHello();
    }
}

File: app/src/main/java/com/androidskeleton/module/ui/home/HomeContract.kt
package com.androidskeleton.module.ui.home

import com.androidskeleton.module.base.BasePresenter
import com.androidskeleton.module.base.BaseView

interface HomeContract {
    interface View : BaseView<Presenter?> {
        fun greetings(message: String?)
    }

    interface Presenter : BasePresenter<View?> {
        fun sayHello()
    }
}